cmd/mt-kafka-mdm-report-out-of-order/inputooofinder.go

```diff
@@ -0,0 +1,147 @@
+package main
+
+import (
+	"os"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/grafana/metrictank/idx/cassandra"
+	"github.com/grafana/metrictank/schema"
+	"github.com/grafana/metrictank/schema/msg"
+	log "github.com/sirupsen/logrus"
+)
+
+type Track struct {
+	Name string
+	Tags []string
```

Comment:
You'll also need the metric interval here to track reordering. *Ideally* you would have an actual reorderbuffer to do the reorder logic for you.

Reply:
Do you think it's worth the overhead in this case?

Reply:
It would help to provide the exact same values we are seeing from Metrictank. Bifrost does not mimic this logic, so it's possible that out-of-order *and* duplicate datapoints get through bifrost and are being sent to Metrictank.

For example, a datapoint `d1` is sent at `T` and then `d2` at `T+30` then `d3` at `T` (again). Even if all of these go to the same bifrost instance (likely if partitioned) bifrost will detect `d3` as *slightly* out of order not as a duplicate (because like this current set of code, it only tracks most recent datapoint). Bifrost will allow `d3` through because it could be reorderable. MT will reject `d3` as a duplicate. It would be ideal if this code could detect that case for us. If we find that this is the cause for most of that 2% extra datapoints we see, then if might be worth adding this logic to bifrost (at the cost of more memory/complexity).

Reply:
You were spot on with this -- with a reorder window >1, all of the "out of order" metrics I was seeing in my stage testing turned out to be duplicates.